tests/foreman/api/test_user.py

```diff
@@ -675,3 +677,255 @@ def test_positive_info_users_ssh_key(self):
+            base_dn=settings.ldap.basedn,
+            groups_base=settings.ldap.grpbasedn,
+            attr_firstname=LDAP_ATTR['firstname'],
+            attr_lastname=LDAP_ATTR['surname'],
+            attr_login=LDAP_ATTR['login_ad'],
+            server_type=LDAP_SERVER_TYPE['API']['ad'],
+            attr_mail=LDAP_ATTR['mail'],
+            name=gen_string('alpha'),
+            host=settings.ldap.hostname,
+            tls=False,
+            port='389',
+            location=[cls.loc],
+            organization=[cls.org],
+        ).create()
+
+    def tearDown(self):
+        for user in entities.User().search(query={'search': u'login={}'.format(
+                self.ldap_user_name)}):
+            user.delete()
+        super(ActiveDirectoryUserTestCase, self).tearDown()
+
+    @tier2
+    @upgrade
+    def test_positive_create_in_ldap_mode(self):
+        """Create User in ldap mode
+
+        :id: 6f8616b1-5380-40d2-8678-7c4434050cfb
+
+        :expectedresults: User is created without specifying the password
+
+        :CaseLevel: Integration
+        """
+        for username in valid_usernames_list():
+            with self.subTest(username):
+                user = entities.User(
+                    login=username,
+                    auth_source=self.authsource,
+                    password='',
+                ).create()
+                self.assertEqual(user.login, username)
+
+    @tier3
+    @skip_if_not_set('ldap')
+    def test_positive_ad_basic_no_roles(self):
+        """Login with LDAP Auth- AD for user with no roles/rights
+
+        :id: 3910c6eb-6eff-4ab7-a50d-ba40f5c24c08
+
+        :setup: assure properly functioning AD server for authentication
+
+        :steps: Login to server with an AD user.
+
+        :expectedresults: Log in to foreman successfully but cannot access entities.
+
+        :CaseLevel: System
+        """
+        sc = ServerConfig(
+            auth=(self.ldap_user_name, self.ldap_user_passwd),
+            url=self.sat_url,
+            verify=False
+        )
+        with self.assertRaises(HTTPError):
+            entities.Architecture(sc).search()
+
+    @tier3
+    @upgrade
+    @skip_if_not_set('ldap')
+    def test_positive_access_entities_from_ldap_org_admin(self):
+        """LDAP User can access resources within its taxonomies if assigned
+        role has permission for same taxonomies
+
+        :id: 522063ad-8d39-4f05-bfd3-dfa0fa73f4e1
+
+        :steps:
+
+            1. Create Org Admin and assign taxonomies to it
+            2. Create LDAP user with same taxonomies as role above
+            3. Assign Org Admin role to user above
+            4. Login with LDAP user and attempt to access resources
+
+        :expectedresults: LDAP User should be able to access all the resources
+            and permissions in taxonomies selected in Org Admin role
+
+        :CaseLevel: System
+        """
+        name = gen_string('alpha')
```

Comment:
Confuses with username or rolename
```suggestion
        role_name = gen_string('alpha')
```